Dockerfile

```diff
@@ -4,27 +4,30 @@ RUN apt update && apt -y upgrade && \
   apt install -y gnupg2 wget curl && \
   # llvm
   wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | gpg --dearmor > /usr/share/keyrings/llvm-snapshot.gpg && \
-  echo 'deb [signed-by=/usr/share/keyrings/llvm-snapshot.gpg] http://apt.llvm.org/bookworm/ llvm-toolchain-bookworm-17 main' >> /etc/apt/sources.list && \
+  echo 'deb [signed-by=/usr/share/keyrings/llvm-snapshot.gpg] http://apt.llvm.org/bookworm/ llvm-toolchain-bookworm-18 main' >> /etc/apt/sources.list && \
   apt update && \
-  apt install -y libllvm-17-ocaml-dev libllvm17 llvm-17 llvm-17-dev llvm-17-doc llvm-17-examples llvm-17-runtime \
-  clang-17 clang-tools-17 clang-17-doc libclang-common-17-dev libclang-17-dev libclang1-17 clang-format-17 python3-clang-17 clangd-17 clang-tidy-17 \
-  libpolly-17-dev \
-  libfuzzer-17-dev \
-  lldb-17 \
-  lld-17 \
-  libc++-17-dev libc++abi-17-dev \
-  libomp-17-dev \
-  libclc-17-dev \
-  libunwind-17-dev \
-  libmlir-17-dev mlir-17-tools \
+  apt install -y libllvm-18-ocaml-dev libllvm18 llvm-18 llvm-18-dev llvm-18-doc llvm-18-examples llvm-18-runtime \
+  clang-18 clang-tools-18 clang-18-doc libclang-common-18-dev libclang-18-dev libclang1-18 clang-format-18 python3-clang-18 clangd-18 clang-tidy-18 \
+  libpolly-18-dev \
+  libfuzzer-18-dev \
+  lldb-18 \
+  lld-18 \
+  libc++-18-dev libc++abi-18-dev \
+  libomp-18-dev \
+  libclc-18-dev \
+  libunwind-18-dev \
+  libmlir-18-dev mlir-18-tools \
```

Comment:
**Security Issue**: Upgrading to LLVM 18 packages without validating compatibility and security patches may introduce vulnerabilities or compatibility issues with the rest of the system. <br> **Fix**: Ensure all LLVM 18 packages are compatible with the application and do not introduce security vulnerabilities. Review the LLVM 18 changelog for security patches and potentially breaking changes. <br> **Code Suggestion**: 
 ```
 -apt install -y libllvm-17-ocaml-dev libllvm17 llvm-17 llvm-17-dev llvm-17-doc llvm-17-examples llvm-17-runtime \
 -clang-17 clang-tools-17 clang-17-doc libclang-common-17-dev libclang-17-dev libclang1-17 clang-format-17 python3-clang-17 clangd-17 clang-tidy-17 \
 -libpolly-17-dev \
 -libfuzzer-17-dev \
 -lldb-17 \
 -lld-17 \
 -libc++-17-dev libc++abi-17-dev \
 -libomp-17-dev \
 -libclc-17-dev \
 -libunwind-17-dev \
 -libmlir-17-dev mlir-17-tools
 +apt install -y libllvm-18-ocaml-dev libllvm18 llvm-18 llvm-18-dev llvm-18-doc llvm-18-examples llvm-18-runtime \
 +clang-18 clang-tools-18 clang-18-doc libclang-common-18-dev libclang-18-dev libclang1-18 clang-format-18 python3-clang-18 clangd-18 clang-tidy-18 \
 +libpolly-18-dev \
 +libfuzzer-18-dev \
 +lldb-18 \
 +lld-18 \
 +libc++-18-dev libc++abi-18-dev \
 +libomp-18-dev \
 +libclc-18-dev \
 +libunwind-18-dev \
 +libmlir-18-dev mlir-18-tools
 ```